RecoLocalTracker/SiPhase2VectorHitBuilder/src/VectorHitBuilderAlgorithm.cc

```diff
@@ -0,0 +1,574 @@
+#include "RecoLocalTracker/SiPhase2VectorHitBuilder/interface/VectorHitBuilderAlgorithm.h"
+#include "Geometry/CommonDetUnit/interface/GeomDet.h"
+#include "Geometry/CommonTopologies/interface/PixelTopology.h"
+#include "DataFormats/TrackerRecHit2D/interface/VectorHit2D.h"
+
+bool VectorHitBuilderAlgorithm::LocalPositionSort::operator()(Phase2TrackerCluster1DRef clus1,
+                                                              Phase2TrackerCluster1DRef clus2) const {
+  static std::map<std::pair<Phase2TrackerCluster1DRef, Phase2TrackerCluster1DRef>, bool> cache;
+
+  if (cache.find(std::make_pair(clus1, clus2)) != cache.end())
+    return cache[std::pair(clus1, clus2)];
+  else if (cache.find(std::make_pair(clus2, clus1)) != cache.end())
+    return !cache[std::pair(clus1, clus2)];
+
+  const PixelGeomDetUnit* gdu1 = dynamic_cast<const PixelGeomDetUnit*>(geomDet_);
+  auto&& lparams1 = cpe_->localParameters(*clus1, *gdu1);  // x, y, z, e2_xx, e2_xy, e2_yy
+  auto&& lparams2 = cpe_->localParameters(*clus2, *gdu1);  // x, y, z, e2_xx, e2_xy, e2_yy
+
+  cache[std::make_pair(clus1, clus2)] = lparams1.first.x() < lparams2.first.x();
+  return lparams1.first.x() < lparams2.first.x();
+  ;
+}
+
+void VectorHitBuilderAlgorithm::run(edm::Handle<edmNew::DetSetVector<Phase2TrackerCluster1D>> clusters,
+                                    VectorHitCollectionNew& vhAcc,
+                                    VectorHitCollectionNew& vhRej,
+                                    edmNew::DetSetVector<Phase2TrackerCluster1D>& clustersAcc,
+                                    edmNew::DetSetVector<Phase2TrackerCluster1D>& clustersRej) const {
+  LogDebug("VectorHitBuilderAlgorithm") << "Run VectorHitBuilderAlgorithm ... \n";
+  const edmNew::DetSetVector<Phase2TrackerCluster1D>* ClustersPhase2Collection = clusters.product();
+
+  std::unordered_map<DetId, std::vector<VectorHit>> tempVHAcc, tempVHRej;
```

Comment:
not needed (none of the two)

Reply:
Is it not clear to me how I would go about removing these maps?

Reply:
I propose, let's make a first release that reproduces earlier results, i.e. is correct.
Optimization will come in a second PR (at least one can then check regression)
btw how this PR will be validated (besides not affecting current WF?)

Reply:
We know the expected behavior from studies done for the PR last year (for example https://jschulte.web.cern.ch/jschulte/VH/plots/plots_highPurity/effandfakePtEtaPhi.pdf) were we see how the effciency drops when VHs are activated (which will be recovered with the upcoming configuration changes.) I can repeat these kind of studies with the current version of the code. 

I got the impression you wanted the performance tuned before we integrate this, but if we can postpone that for a follow-up that would certainly make things easier for me.